code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/VolumeDaoImpl.java

```diff
@@ -96,4 +98,20 @@ public void createVolumeInStoragePool(Map<String, Object> volumeData, String vol
 
         return volumes;
     }
+
+    @Override
+    public Set<? extends Volume> findNonremovedVolumesWithNoOtherMounts(long instanceId) {
+        List<VolumeRecord> vols = create()
+            .selectDistinct(VOLUME.fields())
+            .from(MOUNT)
+            .join(VOLUME)
+                .on(VOLUME.ID.eq(MOUNT.VOLUME_ID).and(VOLUME.REMOVED.isNull()))
+            .where(MOUNT.VOLUME_ID.in(
```

Comment:
This is the file @ibuildthecloud referenced:

https://github.com/rancher/cattle/blob/275ff45c25f9a2064af33a21a4e6137b0bcabbd9/code/iaas/agent-instance/src/main/java/io/cattle/platform/agent/instance/service/impl/NicPerVnetNicLookup.java

I'm scarred for life by that bug.